src/com/google/javascript/jscomp/PeepholeFoldConstants.java

```diff
@@ -1064,15 +1064,24 @@ static TernaryValue evaluateComparison(int op, Node left, Node right, boolean us
 
       case Token.STRING:
         if (right.isString()) {
-          // Only eval if they are the same type
+          return compareStrings(op, left, right);
+        } else if (right.isNumber()) {
+          // See http://www.ecma-international.org/ecma-262/6.0/#sec-strict-equality-comparison
           switch (op) {
             case Token.SHEQ:
-            case Token.EQ:
-              return areStringsEqual(left.getString(), right.getString());
-
+              return TernaryValue.FALSE;
             case Token.SHNE:
-            case Token.NE:
-              return areStringsEqual(left.getString(), right.getString()).not();
+              return TernaryValue.TRUE;
+          }
+          return compareAsNumbers(op, IR.number(NodeUtil.getNumberValue(left)), right, useTypes);
```

Comment:
Isn't this whole
`IR.number(NodeUtil.getNumberValue(x))`
extraneous with what compareAsNumbers does anyway?